src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -259,6 +260,18 @@ public static bool IsMsBuildBasedProject(string projectFullPath)
             AddPropertyIfHasValue(args, "RestorePackagesPath", packagesDirectory);
             AddPropertyIfHasValue(args, "SolutionDir", solutionDirectory);
 
+            // Disable parallel and use ContinueOnError since this may run on an older
+            // version of MSBuild that do not support SkipNonexistentTargets.
+            // When BuildInParallel is used with ContinueOnError it does not continue in
+            // some scenarios.
+            // Allow opt in to msbuild 15.5 behavior with NUGET_RESTORE_MSBUILD_USESKIPNONEXISTENT
+            var nonExistFlag = Environment.GetEnvironmentVariable("NUGET_RESTORE_MSBUILD_USESKIPNONEXISTENT");
```

Comment:
I'm not in love with the fact that this is entirely opt-in.  If there's a smart way for us to detect they're using MSBuild 15, I think it should be on by default and be opt-out.  What do you think?

Reply:
Do you have any ideas on how to get that? Since this is only for nuget.exe I was planning to fix that in a later PR. MSBuild and dotnet will use the 15.5 features by default.

Ideally it would determine this in the target, but the only property I could find was for `15.0` and it did not include the minor version.

NuGet could potentially probe the msbuild.exe it finds to read the assembly version, it might be worth the cost of this.